src/editor/blocks/visibility-attribute/index.js

```diff
@@ -0,0 +1,84 @@
+/**
+ * External dependencies
+ */
+import { assign } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Fragment } from '@wordpress/element';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import { InspectorControls } from '@wordpress/block-editor';
+import { PanelBody, SelectControl } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+const ATTRIBUTE_NAME = 'newsletterVisibility';
+
+const visibilityOptions = [
+	{
+		label: __( 'Email and Web', 'newspack-newsletters' ),
+		value: '',
+	},
+	{
+		label: __( 'Email only', 'newspack-newsletters' ),
+		value: 'email',
+	},
+	{
+		label: __( 'Web only', 'newspack-newsletters' ),
+		value: 'web',
+	},
+];
+
+const addVisibilityAttribute = settings => {
+	settings.attributes = assign( settings.attributes, {
+		[ ATTRIBUTE_NAME ]: {
+			type: 'string',
+			default: visibilityOptions[ 0 ].value,
+		},
+	} );
+	return settings;
+};
+
+const withVisibilityControl = createHigherOrderComponent( BlockEdit => {
+	return props => {
+		const { attributes, setAttributes } = props;
+		const value = attributes[ ATTRIBUTE_NAME ];
+		return (
+			<Fragment>
+				<BlockEdit { ...props } />
```

Comment:
WDYT about adding a notice above the component – if it's web- or email-only? With a more sophisticated layout, clicking each block to verify the sidebar setting is correct would be cumbersome.

Reply:
I'm not sure – it's a tough cookie. This feature has a lot of impact on the WYSIWYG aspect of the editing and I see a lot of potential for user error. Someone might mark the block as web-only, forget about it, and wonder why something did not make it the recipient. This is obviously more problematic with web-only, since you can't edit an already sent email. I think the layout disruption is less important compared to the potential for "missing" content. 
WDYT @thomasguillot ?

Reply:
This is really tricky because it can be extremely complex and it can easily look incredibly busy.

I was thinking, can we append the visibility status to the list view?

![Screenshot 2021-10-13 at 09 33 00](https://user-images.githubusercontent.com/177929/137097288-4d83f2f5-bbde-4cee-a468-201f50eb74f6.png)

Reply:
This is an attempt to be as frictionless as possible. The all-around border is so it can visually cover grouped blocks and the label is `position:absolute;` so it doesn't interfere with different layouts.

https://user-images.githubusercontent.com/820752/137503284-a791714c-f5ad-4e55-a75e-04f2d125140d.mov

Reply:
> This is an attempt to be as frictionless as possible. [...]

This is quite nice. Are you using box-shadow? I can see the content slightly shifting when it doesn't have the yellow border.

Reply:
Nice ✌️